main/ts_client.c

```diff
@@ -86,16 +92,24 @@ char *exec_or_create(char *node)
 
 void ts_parse_uri(const char *uri, TSUriElems *params)
 {
+    params->ts_list_subnodes = -1;
+    params->ts_device_id = NULL;
+    params->ts_target_node = NULL;
+
     if (uri == NULL || uri[0] == '\0') {
+        #ifndef UNIT_TEST
         ESP_LOGE(TAG, "Got invalid uri");
+        #endif
```

Comment:
I suggest to mock the ESP_LOGE and so on commands similar to [charge-controller zephyr.h mocks](https://github.com/LibreSolar/charge-controller-firmware/blob/master/test/zephyr.h). Avoids lots of disturbing #ifdefs in the middle of the code.

Reply:
done :)